main.cpp

```diff
@@ -195,6 +199,38 @@ void mergeImages()
   }
 }
 
+void FlipImage()
+{
+  cout << "Please choose\n"
+          "1->To flip the image horizontally\n"
+          "2->To flip the image vertically\n"
+          "your choice is: ";
+  int choice;
+  cin >> choice;
+  // flip horizontally
+  if (choice == 1)
+  {
+    for (int i = 0; i < SIZE; i++)
+    {
+      for (int j = 0; j < 127; j++)
+      {
+        swap(image[i][j], image[i][255-j]);
+      }
+    }
+  }
+  // flip vertically
+  else if (choice == 2)
+  {
+    for (int j = 0; j < SIZE; j++)
+    {
+      for (int i = 0; i < 127; i++)
+      {
+        swap(image[i][j], image[255-i][j]);
+      }
+    }
+  }
+}
+
```

Comment:
Here try using one nested loop that has the conditions inside of it instead of having two nested loops with the conditions outside
```suggestion
 for(int i = 0; i < SIZE; i++)
 {
     for(int j = 0; j < SIZE; j++)
     {
          if(choice == 1)
          {
              swap(image[i][j], image[i][255-j]);
          }
          else  
          {
              swap(image[i][j], image[255-i][j]);
          }
     }
 }
}
```

Reply:
@mahmoudsamyhamed, I agree with you on the concept, but I don't think the code you added has the same behavior as Mohamed's original code.

@NASSERMO2004 Can you try to use a single loop instead and make sure that it's still working as expected?

Reply:
@mahmoudsamyhamed @NASSERMO2004 We'll look into it again later, for now let's get this merged.